### PR TITLE
url: add the authority method

### PR DESCRIPTION
**Pull request by unleashed**
This is a revisit of #15 to add a method to be able to get the authority on its own.

Much has changed in the meantime, and the method can now be implemented without needing an additional struct.

### CODE REVIEW

File: url/src/lib.rs
Comment (djc) [inline url/src/lib.rs:686]: Are you sure that this documentation is actually how it works? Would be good to add a unit test that covers some of these cases, with at least one IDNA-encoded domain, one percent-encoded host and an IPv6 host.

**Review by djc — COMMENTED** [review]
Thanks for working on this, I think it makes sense to add this. Before we merge it, I'd look to see a little more test coverage to ensure the current documentation matches the implementation.

File: url/src/lib.rs
Comment (unleashed) [reply url/src/lib.rs:686]: Thanks for thr review, @djc.

Yes, I am sure - the documentation is taken from methods that already state this and are implemented in a similar fashion by taking an internal slice. I've added another commit with a few more doc tests to exercise what is being said here and also added a note on ports for special URLs, which are omitted from the authority if they match the well known scheme port.

Comment (djc) [reply url/src/lib.rs:686]: Thanks for adding the tests!

I'd prefer to have this as unit tests not doctests, otherwise the example code gets a bit too bulky for my taste.

Comment (unleashed) [reply url/src/lib.rs:686]: Agreed, it's a bit too much for docs. Moved those to unit tests leaving just a few examples and also took the opportunity to add unit tests for `has_authority`.

**Review by djc — APPROVED** [review]

File: url/src/lib.rs
Comment (djc) [reply url/src/lib.rs:686]: Thanks!

**Comment by unleashed** [comment]
Rebased to fix conflicts.

**Comment by SimonAdameit** [comment]
Hi, I'd be interested in having this. What's still required to get it merged?

**Comment by unleashed** [comment]
I don't know of any specific roadblock other than having a maintainer review and approve it.

**Comment by pehala** [comment]
I would like to revive this PR, is there anything required other than rebasing to the current main?

**Comment by valenting** [comment]
Rebase on the latest changes for the CI to run.
Then we can probably merge this.

**Comment by unleashed** [comment]
@pehala @valenting rebased on top of latest master.

**Review by valenting — APPROVED** [review]
Thanks!

**Pull request merged**
